interceptor/tracing_interceptor.go

```diff
@@ -357,7 +368,20 @@ func (t *tracingActivityInboundInterceptor) ExecuteActivity(
 
 type tracingWorkflowInboundInterceptor struct {
 	WorkflowInboundInterceptorBase
-	root *tracingInterceptor
+	root        *tracingInterceptor
+	spanCounter uint16
+	info        *workflow.Info
+}
+
+// newIdempotencyKey returns a new idempotency key by incrementing the span counter and interpolating
+// this new value into a string that includes the workflow namespace/id/run id and the interceptor type.
+func (t *tracingWorkflowInboundInterceptor) newIdempotencyKey() string {
+	t.spanCounter++
+	return fmt.Sprintf("WorkflowInboundInterceptor:%s:%s:%s:%d",
```

Comment:
We might want to consider a shared counter across all workflow interceptors instead. However including the "interceptor type" prefix ensures that we can add an independent counter to `tracingWorkflowOutboundInterceptor` later if we want to without worrying about idempotency key collisions.

Reply:
Yeah, I think what is present now is reasonable. It is much easier to customize/expose this later if this abstraction is doubly used than to overthink it now.